enclave/src/lib.rs

```diff
@@ -939,6 +974,14 @@ pub fn scan_block_for_relevant_xt(block: &Block) -> SgxResult<Vec<OpaqueCall>> {
             }
         }
     }
+    // TODO KSR :- Create Failed extrinsic management mechanism
+    // Intial Idea:-
+    // O1-Check content of pending set.
+    // It should have nothing in ideal case.
+    // But if there is any elemnt left
+    //
+    // then send it again and move it to active  set
+    // Move everthing from active set to pending set.
```

Comment:
I don't think the scan_block_for_relevant_xt function is the right place for this. Here we're searching for extrinsics that should be executed within the enclave.
What you're doing is somewhat similiarly implemented at https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/master/enclave/src/lib.rs#L377-L379
maybe implement a similar function and place it beneath or above this function?